File: src/main/java/com/negosud/api/service/CommandeFournisseurService.java
package com.negosud.api.service;

import com.negosud.api.dto.request.CommandeFournisseurRequestDto;
import com.negosud.api.dto.request.StatutCommandeDto;
import com.negosud.api.dto.response.CommandeFournisseurResponseDto;

import java.util.List;

public interface CommandeFournisseurService {

    List<CommandeFournisseurResponseDto> findAll();

    CommandeFournisseurResponseDto findById(Long id);

    CommandeFournisseurResponseDto create(CommandeFournisseurRequestDto dto);

    CommandeFournisseurResponseDto update(Long id, CommandeFournisseurRequestDto dto);

    /**
     * Change le statut de la commande.
     * Si le statut passe à LIVREE, le stock des articles est mis à jour automatiquement.
     */
    CommandeFournisseurResponseDto changerStatut(Long id, StatutCommandeDto dto);

    void delete(Long id);

    /**
     * Crée automatiquement une commande fournisseur de réapprovisionnement pour un article
     * dont le stock est passé sous le seuil minimum.
     */
    void creerReapprovisionnementAuto(com.negosud.api.model.entity.Article article);
}

File: front/src/services/commandeFournisseurService.js
import api from './api';

const BASE = '/commandes-fournisseurs';

export const commandeFournisseurService = {
  findAll: () => api.get(BASE),
  findById: (id) => api.get(`${BASE}/${id}`),
  create: (data) => api.post(BASE, data),
  update: (id, data) => api.put(`${BASE}/${id}`, data),
  changerStatut: (id, statut) => api.patch(`${BASE}/${id}/statut`, { statut }),
  ajouterLigne: (id, data) => api.post(`${BASE}/${id}/lignes`, data),
  supprimerLigne: (id, ligneId) => api.delete(`${BASE}/${id}/lignes/${ligneId}`),
  delete: (id) => api.delete(`${BASE}/${id}`),
};
